code/game/machinery/bots/farmbot.dm

```diff
@@ -190,6 +190,15 @@
 	spawn(150)
 		mode = 0
 
+/obj/machinery/bot/farmbot/emag_ai(mob/living/silicon/ai/A)
```

Comment:
same with all these bots, more maintainable anyway if someone changes emag behavior (most of it could be done at the bot level too)